CMakeLists.txt

```diff
@@ -372,20 +365,24 @@ if (HCC_INTEGRATE_ROCDL)
   set(ROCDL_BUILD_DIR "${PROJECT_BINARY_DIR}/rocdl")
   file(MAKE_DIRECTORY ${ROCDL_BUILD_DIR})
 
-  # custom commands to build rocdl
-  add_custom_target(rocdl
-    COMMAND ${CMAKE_COMMAND} ${ROCDL_SRC_DIR}
-            -DLLVM_DIR=${CLANG_BIN_DIR}
-            -DAMDHSACOD=${AMDHSACOD}
-            -DGENERIC_IS_ZERO=ON
-            -DBUILD_HC_LIB=ON
-    COMMAND make -j ${NUM_BUILD_THREADS} # not portable, but we need it this way
-    WORKING_DIRECTORY ${ROCDL_BUILD_DIR}
-    DEPENDS clang_links llvm-link
-  )
+  set(LLVM_DIR ${CLANG_BIN_DIR})
+  set(AMDHSACOD ${AMDHSACOD})
+  set(GENERIC_IS_ZERO ON)
+  set(BUILD_HC_LIB ON)
+  set(AMDGCN_TARGETS_LIB_LIST "AMDGCN_LIB_TARGETS")
+  find_package(LLVM REQUIRED CONFIG PATHS ${CMAKE_BINARY_DIR}/compiler NO_DEFAULT_PATH)
+
+  list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
+  include(AddLLVM)
+
+  add_definitions(${LLVM_DEFINITIONS})
+
+  # build rocdl
+  add_subdirectory(${ROCDL_SRC_DIR})
 
   file(MAKE_DIRECTORY ${ROCDL_BUILD_DIR}/lib)
-  add_custom_command(TARGET rocdl POST_BUILD
+  add_custom_target(rocdl_links DEPENDS ${AMDGCN_LIB_TARGETS})
```

Comment:
the library we include in hcc has parameter `AMDGCN_TARGETS_LIB_LIST` that contains variable name to return libs targets list: `set(AMDGCN_TARGETS_LIB_LIST "AMDGCN_LIB_TARGETS")` so after `add_subdirectory(${ROCDL_SRC_DIR})` we have all target names built by library in `AMDGCN_LIB_TARGETS` variable